docs/source/api/data-management/graphs.rst

```diff
@@ -0,0 +1,262 @@
+
+.. _graph_programming_interface:
+
+---------------------
+Programming interface
+---------------------
+
+All types and functions in this section are declared in the
+``oneapi::dal::preview`` namespace and are available via inclusion of the
+``oneapi/dal/graph/common.hpp`` header file.
+
+Graph
+-----
+
+The :txtref:`graph` concept is represented by the types with the ``_graph`` suffix and all of them are
+:term:`reference-counted <Reference-counted object>`:
+
+1. The instance stores pointers to the graph :capterm:`topology`
+   and :capterm:`attributes <Attribute>` of vertices and edges.
+
+2. The reference count indicating how many graph objects refer to the same implementation.
+
+3. The graph increments the reference count
+   for it to be equal to the number of graph objects sharing the same implementation.
+
+4. The graph decrements the reference count when the
+   graph goes out of the scope. If the reference count is zero, the graph
+   frees its implementation.
+
+.. _api_graph_template_type:
+
+The :txtref:`graph` types are defined as templated classes:
+
+.. code-block:: cpp
+
+  template <typename VertexValue,
+            typename EdgeValue,
+            typename GraphValue,
+            typename IndexType,
+            typename Allocator>
+  class [graph_name]_graph;
+
+.. .. namespace:: oneapi::dal::preview
+.. .. struct:: graph_traits
+
+.. list-table:: 
+   :widths: 20 40 40 
+   :header-rows: 1
+
+   * - Type name
+     - Description
+     - Supported types
+   * - ``VertexValue``
+     - Type of the vertex :capterm:`attribute <Attribute>` values
+     - :ref:`Empty value <api_empty_value>`
+   * - ``EdgeValue``
+     - Type of the edge :capterm:`attribute <Attribute>` values
+     - ``std::int32``, ``double``, :ref:`Empty value <api_empty_value>`  
+   * - ``GraphValue``
+     - Type of the graph :capterm:`attribute <Attribute>` value
+     - :ref:`Empty value <api_empty_value>`  
+   * - ``IndexType``
+     - Type of the :capterm:`vertex indices <Vertex index>`
+     - ``std::int32`` 
+   * - ``Allocator``
+     - graph allocator type.
+     - C++17 (ISO/IEC 14882:2017) compliant allocator
+
+
+:ref:`Empty value <api_empty_value>` tag structure is used to define 
+the absence of specified attribute of graph.
+
+.. _api_empty_value:
+
+.. onedal_class:: oneapi::dal::preview::empty_value
+
+Graph class contains default and move constructor, assignment.
+The graph is accessed using the :ref:`service functions <api_graph_service>`.
+
+.. list-table:: 
+  :widths: 30 70 
+  :header-rows: 1
+
+  * - ``graph_type`` method
+    - Description
+  * - Default constructor
+    - Constructs an empty graph object 
+  * - Move constructor
+    - Swaps implementation  
+  * - Move assignment
+    - Swaps implementation 
+
+
+.. _api_graph_traits:
+
+Graph traits
+------------
+   
+Graph traits is a data type that defines the data model, a set of types
+associated with the graph, used by processing and service functionality.
```

Comment:
`used by processing and service functionality` what this refers to?

Reply:
You mean that some link is required here? If yes, processing is not available in docs right now(algos). I can add link for service.

Reply:
```suggestion
associated with the graph, used by processing and :txtref:`service functionality <api_graph_service>`.
```

Reply:
no I didn't mean it as in link :) sorry

I meant: that part of the sentence seems disconnected from the beginning of the sentence. It's not clear what exactly is used by processing and service functionality. The data model? 
Also: is `a set of types associated with the graph` a definition of a data model?